src/components/modals/RequestPermissionsModal.tsx

```diff
@@ -0,0 +1,76 @@
+import * as React from 'react'
+import { AirshipBridge } from 'react-native-airship'
+
+import s from '../../common/locales/strings'
+import { Checkbox } from '../themed/Checkbox'
+import { MainButton } from '../themed/MainButton'
+import { ModalCloseArrow } from '../themed/ModalParts'
+import { ThemedModal } from '../themed/ThemedModal'
+import { MessageText, TitleText } from '../themed/ThemedText'
+
+interface Props {
+  bridge: AirshipBridge<ChoiceMap>
+  choices: ChoiceMap
+  message: string
+}
+
+export interface ChoiceMap {
```

Comment:
I would call this `PermissionsModalResult`, since a "choice map" could be anything.